src/CSET/operators/collapse.py

```diff
@@ -163,10 +181,35 @@ def collapse_by_hour_of_day(
     To apply this operator successfully there must only be one time dimension.
     Should a MultiDim exception be raised the user first needs to apply the
     collapse operator to reduce the time dimensions before applying this
-    operator.
+    operator. If multi_case is true the collapse_by_lead_time operator is
+    applied and performs this step.
     """
     if method == "PERCENTILE" and additional_percent is None:
         raise ValueError("Must specify additional_percent")
+    if (
+        isinstance(cube, iris.cube.Cube)
+        and is_time_aggregatable(cube)
+        and not multi_case
+    ):
+        raise TypeError(
+            "multi_case must be true for a cube containing two time dimensions"
+        )
+    if (
+        isinstance(cube, iris.cube.Cube)
+        and not is_time_aggregatable(cube)
+        and multi_case
+    ):
+        raise TypeError(
+            "multi_case must be false for a cube containing one time dimension"
+        )
+    if isinstance(cube, iris.cube.CubeList) and not multi_case:
```

Comment:
```suggestion
    elif isinstance(cube, iris.cube.CubeList) and not multi_case:
```
These conditions are mutually exclusive, so we only need to check them if the previous one didn't match.